lib/kafka/async_producer.rb

```diff
@@ -74,6 +74,8 @@ def initialize(sync_producer:, max_queue_size: 1000, delivery_threshold: 0, deli
       raise ArgumentError unless max_queue_size > 0
       raise ArgumentError unless delivery_threshold >= 0
       raise ArgumentError unless delivery_interval >= 0
+      raise ArgumentError unless max_retries >= 0
+      raise ArgumentError unless retry_backoff >= 0
```

Comment:
These checks were in our fork but not in the official version. We backport them to keep existing behavior.